### PR TITLE
[4.0] Fixing switcher label for true/false

### PR DESCRIPTION
**Pull request by Quy**
Pull Request for Issue #26749.

### Summary of Changes
In Global Configuration, `0/1` are saved as `true/false`. Read https://github.com/joomla/joomla-cms/issues/26749#issuecomment-548873881
`false` cast as string returns empty string and `true` returns 1.
Assign it `0` so it can match the option value `0`.
```
			<option value="0">JNO</option>
			<option value="1">JYES</option>
```

### Testing Instructions
Go to Global Configurations
See Site Offline missing label
Apply PR
Make sure switcher labels not missing

Open configuration.php
Toggle Site Offline
Make sure `$offline` is saved as true or false

### CODE REVIEW

File: libraries/src/Form/Field/RadioField.php [hunk 1]
```diff
@@ -94,6 +94,12 @@ protected function getLayoutData()
 	{
 		$data = parent::getLayoutData();
 
+		// Casting false as string returns an empty string so assign it 0
+		if (is_bool($this->value) && $this->value === false)
```
Comment (SharkyKZ) [inline libraries/src/Form/Field/RadioField.php]: `is_bool()` check is not necessary.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on f17349d306aea36759e8b1e69d2d724c55e525c5<br /><br />Toggled Site Offline. configuration.php > public $offline=true;<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27098">issues.joomla.org/tracker/joomla-cms/27098</a>.</sub>

**Comment by vaibhavsTekdi** [comment]
I have tested this item :white_check_mark: successfully on f17349d306aea36759e8b1e69d2d724c55e525c5<br /><br />Working fine as per the mentioned "Testing instructions".<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27098">issues.joomla.org/tracker/joomla-cms/27098</a>.</sub>

**Comment by SharkyKZ** [comment]
RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27098">issues.joomla.org/tracker/joomla-cms/27098</a>.</sub>

**Comment by infograf768** [comment]
@HLeithner 
is that OK for you?

**Comment by HLeithner** [comment]
I'm still not sure if this is the proper way to do this. But yeah if it's the best solution to resolve the problem. Shouldn't this be done in the model/formfield layer and not in the layout?

**Comment by HLeithner** [comment]
Forget my comment I mixed 2 pr's it seams to be ok.

**Comment by wilsonge** [comment]
In this specific case wondering if we actually should have this in the form. normally this is correct. However in the normal (non-switcher) radio field it *may* not be.

**Comment by Quy** [comment]
It is working now for switcher and non-switcher.

File: libraries/src/Form/FormField.php [hunk 2]
```diff
@@ -1215,6 +1215,12 @@ protected function getLayoutData()
 
 		$alt = preg_replace('/[^a-zA-Z0-9_\-]/', '_', $this->fieldname);
 
+                // Casting false as string returns an empty string so assign it 0
```
Comment (infograf768) [inline libraries/src/Form/FormField.php]: spces -> tabs

**Comment by HLeithner** [comment]
Changing this in the Jformfield would mean that this is valid for all abstracted classes, that means if I have a input field I get a '0' instead of an empty string correct? (if the function is not overridden)

I wouldn't expect this.

**Comment by Quy** [comment]
It is checking for a `false` value. Empty string does not apply. 
To test, apply PR. See `Path to Cache Folder` field where it is empty. It does not display `0`.

**Comment by HLeithner** [comment]
thats true but if the value is false it would convert it also for a text field into '0' and not ''

**Comment by Quy** [comment]
It is a strict comparison for boolean and false. A string value will not be affected.

**Comment by infograf768** [comment]
@Quy
You mean that when we have:
```
			<option value="true">JYES</option>
			<option value="false">JNO</option> 
```
It would have no impact?

**Comment by Quy** [comment]
I am not at a computer to test but it should not be affected because it is string and not boolean.

**Comment by Quy** [comment]
Also this applies to the configuration file only when storing true/false values if I am not wrong. It has been a while since doing this PR. My memory not so good these days.

**Comment by Fedik** [comment]
I wounder how it may have RTC :wink: 

Doing `if ($this->value === false)` globally in FormField is incorrect, it will affect every field.

That maybe better to do after #27202 will be merged

**Comment by brianteeman** [comment]
Works perfectly for me. I dont see any issues.

@fedik please post exactly where it breaks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27098">issues.joomla.org/tracker/joomla-cms/27098</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 72e779b1a1113e62557bc793340fd363986e9c96<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27098">issues.joomla.org/tracker/joomla-cms/27098</a>.</sub>

**Comment by brianteeman** [comment]
I can't comment about the code but it solves the problem<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27098">issues.joomla.org/tracker/joomla-cms/27098</a>.</sub>

**Comment by Fedik** [comment]
> please post exactly where it breaks

People tested it with switcher field, but this changes:

>  will affect every field

this may lead to unexpected behavior, when example some extension have a custom field, 
which expect value to be boolean `false`, and we force it to be string `'0'`.

**Comment by infograf768** [comment]
Took off RTC until @Fedik comment is taken care of.

**Comment by infograf768** [comment]
setting to pending<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27098">issues.joomla.org/tracker/joomla-cms/27098</a>.</sub>

**Comment by brianteeman** [comment]
@fedik could you please provide a specific example

**Comment by Fedik** [comment]
I do not have a specific example to demonstrate, it a logic assumption that comes from the code review.

If we need this hacky thing, then better to do it specifically for the switcher field, not global.

There a couple ways:
Or move this hacky code to the switcher layout itself `layouts/joomla/form/field/radio/switcher.php`
Or fix these two conditions to work correctly:
https://github.com/joomla/joomla-cms/blob/a9f80d608f0c0e00abf6cdf04c1c4f9932c44c42/layouts/joomla/form/field/radio/switcher.php#L73-L77

**Comment by Quy** [comment]
Is this better?

**Comment by Fedik** [comment]
much better :wink:

**Comment by wilsonge** [comment]
I mean it's grim. But much better than the initial code change. So merging. Thankyou!

**Pull request merged**